src/vm/prestub.cpp

```diff
@@ -639,6 +855,69 @@ PCODE MethodDesc::MakeJitWorker(COR_ILMETHOD_DECODER* ILHeader, DWORD flags, DWO
     LOG((LF_CORDB, LL_EVERYTHING, "MethodDesc::MakeJitWorker finished. Stub is" FMT_ADDR "\n",
          DBG_ADDR(pCode)));
 
+#if defined(FEATURE_JIT_DROPPING)
+    CodeHeader* pCH = ((CodeHeader*)pCode) - 1;
+    _ASSERTE(pCH->GetMethodDesc() == this);
+
+    HeapList* pHL = pCH->GetHeapList();
+
+    if ((CLRConfig::GetConfigValue(CLRConfig::INTERNAL_JitDropEnabled) != 0) &&
+        (CLRConfig::GetConfigValue(CLRConfig::INTERNAL_JitDropMemThreshold) != 0))
+    {
+        if ((totalNCSize - jitDroppedBytes) > CLRConfig::GetConfigValue(CLRConfig::INTERNAL_JitDropMemThreshold) && s_pCalledMethods != NULL)
+        {
+            EX_TRY
+            {
+                // Suspend the runtime.
+                ThreadSuspend::SuspendEE(ThreadSuspend::SUSPEND_OTHER);
+
+                // Walk all other threads.
+                Thread* pThread = nullptr;
+                while ((pThread = ThreadStore::GetThreadList(pThread)) != nullptr)
+                {
+                    pThread->StackWalkFrames(StackWalkCallback, (VOID *)pThread, FUNCTIONSONLY | ALLOW_ASYNC_STACK_WALK);
+                }
+
+                PtrHashMap::PtrIterator i = s_pCalledMethods->begin();
+                while (!i.end())
+                {
+                    MethodDesc *pMD = (MethodDesc *) i.GetValue();
+                    UPTR key = pMD->GetStableHash();
+                    MethodDesc *pFound = (MethodDesc *)s_pExecutedMethods->LookupValue(key, (LPVOID)pMD);
+                    ++i;
+                    if (pFound == (MethodDesc *)INVALIDENTRY)
+                    {
+                        pMD->DropNativeCode();
+                        s_pCalledMethods->DeleteValue(key, (LPVOID)pMD);
+                    }
+                }
+                for (PtrHashMap::PtrIterator i = s_pExecutedMethods->begin(); !i.end();)
+                {
+                    MethodDesc *pMD = (MethodDesc *) i.GetValue();
+                    UPTR key = (UPTR)pMD->GetStableHash();
+                    ++i;
+                    s_pExecutedMethods->DeleteValue(key, (LPVOID)pMD);
+                }
+                ThreadSuspend::RestartEE(FALSE, TRUE);
+            }
+            EX_CATCH
+            {
+            }
+            EX_END_CATCH(SwallowAllExceptions);
+        }
+    }
+#endif
+
+#if defined(FEATURE_JIT_DROPPING)
+    if ((CLRConfig::GetConfigValue(CLRConfig::INTERNAL_JitDropEnabled) != 0) &&
+        (CLRConfig::GetConfigValue(CLRConfig::INTERNAL_JitDropMemThreshold) != 0) &&
+        CLRConfig::GetConfigValue(CLRConfig::INTERNAL_JitDropMethodSizeThreshold) < sizeOfCode &&
+        !IsLCGMethod() && !IsFCall())
+    {
+        LookupOrCreateInCalledMethods(this);
```

Comment:
I'm not quite getting why we have this here and why we have a separate #ifdef for this.